src/LockstakeEngine.sol

```diff
@@ -218,57 +225,49 @@ contract LockstakeEngine is Multicall {
 
     // --- getters ---
 
-    // NOTE: this function will succeed returning the address even if the urn for the specified index hasn't been created yet
-    function getUrn(address owner, uint256 index) external view returns (address urn) {
-        uint256 salt = uint256(keccak256(abi.encode(owner, index)));
-        bytes32 codeHash = keccak256(abi.encodePacked(_initCode()));
-        urn = address(uint160(uint256(
-            keccak256(
-                abi.encodePacked(bytes1(0xff), address(this), salt, codeHash)
-            )
-        )));
-    }
-
-    function isUrnAuth(address urn, address usr) external view returns (bool ok) {
-        ok = _urnAuth(urn, usr);
+    function isUrnAuth(address owner, uint256 index, address usr) external view returns (bool ok) {
+        ok = _urnAuth(owner, _getUrn(owner, index), usr);
```

Comment:
Checking the `urn` existence is important for the `lock` functions in order to avoid to deposit something in a position that doesn't exist. In the prev version, validating the urn existence would require an extra storage load, so it was just implemented in those `lock` functions. Now that we are already consuming the extra sload, IMO just made sense to put it in the checks path everywhere.
I doubted about the consequence of reverting in `isUrnAuth` as well. If we consider this might be a pain in the ass for the UIs we can do it differently there.